app/src/main/java/de/westnordost/streetcomplete/user/LoginFragment.kt

```diff
@@ -110,7 +105,7 @@ class LoginFragment :
     suspend fun hasRequiredPermissions(consumer: OAuthConsumer): Boolean {
         return withContext(Dispatchers.IO) {
             try {
-                val permissionsApi = PermissionsApi(OsmApiModule.osmConnection(consumer))
+                val permissionsApi = PermissionsApi(osmConnection(consumer))
```

Comment:
```suggestion
                /* we didn't save the new OAuthConsumer yet but we want to make an API call with it
                   to check if the user granted all required permissions, this is why we need to 
                   create a new OsmConnection with the supplied consumer instead of using an 
                   injected one */
                val permissionsApi = PermissionsApi(osmConnection(consumer))
```

Had to look up again why this is done. Might as well add a comment here to explain why.